src/components/FileBrowser.vue

```diff
@@ -0,0 +1,294 @@
+        this.requireSession();
+        const request = async (gr, endpoint, params) => {
+          if (params === null) return [];
+          const resp = await gr.get(endpoint, { params });
+          if (resp.status !== 200) return [];
+          return resp.data;
+        };
+        this.rowsLoading = true;
+        const rows = [
+          ...(await request(this.girderRest, GIRDER_FOLDER_ENDPOINT, this.folderParams)),
+          ...(await request(this.girderRest, GIRDER_ITEM_ENDOINT, this.itemParams)),
+        ].map(item => ({
+          name: item.name,
+          type: item._modelType,
+          id: item._id,
+          size: item.size ? this.formatSize(item.size) : '',
+          icon: item._modelType in ICON_MAP ? ICON_MAP[item._modelType] : 'file',
+        }));
+        this.rowsLoading = false;
+        return rows;
+      },
+      watch() {
+        return [
+          this.folderParams,
+          this.itemParams,
+          this.counts,
+          this.location,
+          this.refresh,
+        ];
+      },
+    },
+  },
+  watch: {
+    location(newval, oldval) {
+      // force reset pagination when location changes.
+      if (newval.id !== oldval.id) {
+        this.pagination.page = 1;
+      }
+    },
+    selected(newval) {
+      this.$emit('update:selected', newval);
+    },
+  },
+  computed: {
+    folderParams() {
+      return {
+        parentType: this.location.type,
+        parentId: this.location.id,
+        limit: this.pagination.rowsPerPage,
+        offset: (this.pagination.page - 1) * this.pagination.rowsPerPage,
+      };
+    },
+    itemParams() {
+      const limit = this.counts.nFolders > this.folderParams.offset
+        ? this.pagination.rowsPerPage - (this.counts.nFolders - this.folderParams.offset)
+        : this.pagination.rowsPerPage;
+      const offset = this.folderParams.offset > 0
+        ? this.folderParams.offset - this.counts.nFolders
+        : 0;
+      if (limit <= 0 || (this.location.type !== 'folder')) {
+        return null; // No items should be shown on the current page.
+      }
+      return { folderId: this.location.id, limit, offset };
+    },
+    loading() { return this.rowsLoading || this.breadcrumbLoading; },
+    login() { return this.girderRest.user ? this.girderRest.user.login : ''; },
+    totalItems() { return this.counts.nFolders + this.counts.nItems; },
+  },
+  methods: {
+    toggleAll() {
+      if (this.selected.length === this.rows.length) {
+        this.selected = [];
+      } else {
+        this.selected = this.rows.slice();
+      }
+    },
+    changeLocation(item) {
+      const { type, id, name } = item;
+      if ((this.location.id !== id || this.location.type !== type)
+          && type !== 'item') {
+        this.$emit('update:location', { type, id, name });
+      }
+    },
+    requireSession() {
+      if (!this.login || !this.location) {
+        throw new Error('File Browser expects an active session and a defined location.');
```

Comment:
We should leave open the possibility of browsing anonymously.  As for the `location` prop, we can do a prop validator rather than checking it before every rest call.